R/get.R.pkg.date.R

```diff
@@ -75,9 +96,29 @@ get.R.pkg.date <- function(pkg_vrs, R_vrs) {
 
   start.date <- as.DateYMD(start.date)
   end.date <- as.DateYMD(end.date)
-  as.numeric(end.date - start.date)
-  mid.date <- as.numeric((end.date - start.date)) / 2 + start.date
+
+  # Use day before end-date
+  mid.date <- as.DateYMD((end.date - 1))
+
+  # Check desired versio is being delivered, otherwise change date to 1/1/1970
+  mran.version <- get.mran.version(pkg, mid.date)
+  if (mran.version != vrs) {
+    message1("version mismatch, date set to 1970")
+    return(as.DateYMD("1970-01-01"))
+  }
+
 
 
   return(mid.date)
 } # End.function 2.11
+
+
+
+# Function addded on 2020 05 06
+#' @importFrom utils available.packages
+get.mran.version <- function(pkg, date) {
+  repok <- paste0("https://mran.microsoft.com/snapshot/", date)
+  ap <- as.data.frame(available.packages(repos = repok, type = "binary"))
```

Comment:
The argument `repos` was added in https://github.com/wch/r-source/commit/4a2026e8e0806ba448908ed0a754db75676389e3, which means it's not available before R 3.2

Reply:
great catch.
is there a work around? a quick way to get the packages availalbe on MRAN before 3.2

a brute force solution is to skip MRAN when using R<3.2

Reply:
It might be possible to use the `contriburl` argument to get around this but I'll need to look into it